src/services/studio/studio.service.ts

```diff
@@ -0,0 +1,49 @@
+import {PackageExportTransport} from "../../interfaces/package-export-transport";
+import {packageApi} from "../../api/package-api";
+import {PackageManagerVariableType, PackageWithVariableAssignments} from "../../interfaces/package-manager.interfaces";
+import {dataModelService} from "../package-manager/datamodel-service";
+
+class StudioService {
+
+    public async getExportPackagesWithStudioData(packagesToExport: PackageExportTransport[], withDependencies: boolean) {
+        const studioPackagesWithDataModels = await packageApi.findAllPackagesWithVariableAssignments(PackageManagerVariableType.DATA_MODEL);
+
+        await studioService.setSpaceIdForStudioPackages(packagesToExport, studioPackagesWithDataModels);
+
+        if (withDependencies) {
+            await studioService.setDataModelsForStudioPackages(packagesToExport, studioPackagesWithDataModels);
+        }
+
+        return packagesToExport;
+    }
+
+    private async setSpaceIdForStudioPackages(packages: PackageExportTransport[], studioPackages: PackageWithVariableAssignments[]): Promise<void> {
+        const packageByKey = new Map<string, PackageExportTransport>();
+        packages.forEach(pkg => packageByKey.set(pkg.key, pkg));
```

Comment:
could we make this a pure function by not changing the arguments supplied to the function and instead return new objects